src/mlpack/core/tree/cellbound_impl.hpp

```diff
@@ -982,25 +982,16 @@ template<typename MetricType, typename ElemType>
 template<typename Archive>
 void CellBound<MetricType, ElemType>::serialize(
     Archive& ar,
-    const unsigned int /* version */)
+    std::uint32_t const /* version */)
 {
-  // Allocate memory for the bounds, if necessary.
-  if (Archive::is_loading::value)
-  {
-    if (bounds)
-      delete[] bounds;
-    bounds = new math::RangeType<ElemType>[dim];
-  }
-
-  auto boundsArray = boost::serialization::make_array(bounds, dim);
-  ar & BOOST_SERIALIZATION_NVP(boundsArray);
-  ar & BOOST_SERIALIZATION_NVP(minWidth);
-  ar & BOOST_SERIALIZATION_NVP(loBound);
-  ar & BOOST_SERIALIZATION_NVP(hiBound);
-  ar & BOOST_SERIALIZATION_NVP(numBounds);
-  ar & BOOST_SERIALIZATION_NVP(loAddress);
-  ar & BOOST_SERIALIZATION_NVP(hiAddress);
-  ar & BOOST_SERIALIZATION_NVP(metric);
+  ar & cereal::make_array(bounds, dim);
```

Comment:
We should probably make a wrapper macro for `make_array` like `CEREAL_POINTER_ARRAY()` or something?

Reply:
I think it will be `CEREAL_ARRAY_NVP` :+1: